clients/unshaded/src/main/java/tachyon/client/file/FileSystem.java

```diff
@@ -0,0 +1,365 @@
+   * @throws InvalidPathException if the path is invalid
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  boolean exists(TachyonURI path) throws InvalidPathException, IOException, TachyonException;
+
+  /**
+   * Checks whether a path exists in Tachyon space
+   *
+   * @param path the path in question
+   * @param options options to associate with this operation
+   * @return true if the path exists, false otherwise
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws InvalidPathException if the path is invalid
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  boolean exists(TachyonURI path, ExistsOptions options) throws InvalidPathException, IOException,
+      TachyonException;
+
+  /**
+   * Convenience method to free a path with default options.
+   *
+   * @param path the path to free in Tachyon space
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the given path does not exist
+   * @throws TachyonException if an unexpected tachyon exception is thrown
+   */
+  void free(TachyonURI path) throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Evicts any data under the given path from Tachyon space, but does not delete the data from the
+   * UFS.
+   *
+   * @param path the path to free in Tachyon space
+   * @param options options to associate with this operation
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the given path does not exist
+   * @throws TachyonException if an unexpected tachyon exception is thrown
+   */
+  void free(TachyonURI path, FreeOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Convenience method for get status with default options.
+   *
+   * @param path the path to obtain information about
+   * @return the {@link URIStatus} of the file
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the path does not exist
+   */
+  URIStatus getStatus(TachyonURI path)
+      throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Gets the {@link URIStatus} object that represents the metadata of a Tachyon path.
+   *
+   * @param path the path to obtain information about
+   * @param options options to associate with this operation
+   * @return the {@link URIStatus} of the file
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the path does not exist
+   */
+  URIStatus getStatus(TachyonURI path, GetStatusOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Convenience method for list status with default options.
+   *
+   * @param path the path to list information about
+   * @return a list of {@link URIStatus}s representing the paths which are children of the given
```

Comment:
`URIStatus`s containing information about the files and directories which are children of the given path